TST: Nested arrays failing in array-api-strict git tip

data-apis/array-api-strict#147 broke a bunch of scipy tests.
They need to be sanitized.

CC @ev-br 

```python
$ pip install git+https://github.com/data-apis/array-api-strict.git
$ python dev.py test -b array_api_strict -m array_api_backends
FAILED scipy/integrate/tests/test_cubature.py::TestCubatureProblems::test_scalar_output[array_api_strict-problem13-gk15-1e-05-0.0001] - TypeError: Nested Arrays are not allowed. Use `stack` instead.
FAILED scipy/integrate/tests/test_cubature.py::TestCubatureProblems::test_array_output[array_api_strict-shape0-problem3-gk15-1e-05-0.0001] - TypeError: Nested Arrays are not allowed. Use `stack` instead.
FAILED scipy/ndimage/tests/test_filters.py::TestVectorizedFilter::test_dtype_batch_memory[array_api_strict-False-48-uint8] - TypeError: Nested Arrays are not allowed. Use `stack` instead.
FAILED scipy/stats/tests/test_morestats.py::TestKstat::test_moments_normal_distribution[array_api_strict] - TypeError: Nested Arrays are not allowed. Use `stack` instead.
FAILED scipy/stats/tests/test_morestats.py::TestBoxcox_llf::test_axis[array_api_strict] - TypeError: Nested Arrays are not allowed. Use `stack` instead.
FAILED scipy/stats/tests/test_morestats.py::TestCircFuncs::test_circmean_axis[array_api_strict-circmean] - TypeError: Nested Arrays are not allowed. Use `stack` instead.
FAILED scipy/stats/tests/test_morestats.py::TestCommonAxis::test_axis[array_api_strict-case0] - TypeError: Nested Arrays are not allowed. Use `stack` instead.
FAILED scipy/stats/tests/test_stats.py::test_ttest_ind[array_api_strict] - TypeError: Nested Arrays are not allowed. Use `stack` instead.
FAILED scipy/stats/tests/test_stats.py::TestTTestInd::test_ttest_ind_with_uneq_var[array_api_strict] - TypeError: Nested Arrays are not allowed. Use `stack` instead.
FAILED scipy/stats/tests/test_stats.py::TestJarqueBera::test_axis[array_api_strict] - TypeError: Nested Arrays are not allowed. Use `stack` instead.
FAILED scipy/stats/tests/test_stats.py::TestCombinePvalues::test_monotonicity[array_api_strict-fisher-single] - TypeError: Nested Arrays are not allowed. Use `stack` instead.
```

--- a/scipy/integrate/tests/test_cubature.py
+++ b/scipy/integrate/tests/test_cubature.py
@@ -222,7 +222,9 @@ def _eval_indefinite_integral(F, a, b, xp):
 
     out = 0
     for ind in itertools.product(range(2), repeat=ndim):
-        selected_points = xp.asarray([points[i, j] for i, j in zip(ind, range(ndim))])
+        selected_points = xp.asarray(
+            [float(points[i, j]) for i, j in zip(ind, range(ndim))]
+        )
         out += pow(-1, sum(ind) + ndim) * F(selected_points)
 
     return out

--- a/scipy/ndimage/tests/test_filters.py
+++ b/scipy/ndimage/tests/test_filters.py
@@ -2860,12 +2860,12 @@ class TestVectorizedFilter:
                    if batch_memory == 1 else contextlib.nullcontext())
         with context:
             res = ndimage.vectorized_filter(input, xp.sum, **kwargs)
-            xp_assert_close(res, xp.asarray(ref, dtype=sum_dtype))
+            xp_assert_close(res, xp.asarray(xp.stack(ref), dtype=sum_dtype))
             assert res.dtype == sum_dtype
 
             output = xp.empty_like(input)
             res = ndimage.vectorized_filter(input, xp.sum, output=output, **kwargs)
-            xp_assert_close(res, xp.asarray(ref, dtype=dtype))
+            xp_assert_close(res, xp.asarray(xp.stack(ref), dtype=dtype))
             assert res.dtype == dtype
 
     def test_mode_valid(self, xp):

--- a/scipy/ndimage/tests/test_filters.py
+++ b/scipy/ndimage/tests/test_filters.py
@@ -2860,12 +2860,12 @@ class TestVectorizedFilter:
                    if batch_memory == 1 else contextlib.nullcontext())
         with context:
             res = ndimage.vectorized_filter(input, xp.sum, **kwargs)
-            xp_assert_close(res, xp.asarray(xp.stack(ref), dtype=sum_dtype))
+            xp_assert_close(res, xp.astype(xp.stack(ref), sum_dtype))
             assert res.dtype == sum_dtype
 
             output = xp.empty_like(input)
             res = ndimage.vectorized_filter(input, xp.sum, output=output, **kwargs)
-            xp_assert_close(res, xp.asarray(xp.stack(ref), dtype=dtype))
+            xp_assert_close(res, xp.astype(xp.stack(ref), dtype))
             assert res.dtype == dtype
 
     def test_mode_valid(self, xp):

--- a/scipy/stats/tests/test_morestats.py
+++ b/scipy/stats/tests/test_morestats.py
@@ -1804,7 +1804,7 @@ class TestKstat:
     def test_moments_normal_distribution(self, xp):
         rng = np.random.RandomState(32149)
         data = xp.asarray(rng.randn(12345), dtype=xp.float64)
-        moments = xp.asarray([stats.kstat(data, n) for n in [1, 2, 3, 4]])
+        moments = xp.stack([stats.kstat(data, n) for n in [1, 2, 3, 4]])
 
         expected = xp.asarray([0.011315, 1.017931, 0.05811052, 0.0754134],
                               dtype=data.dtype)
@@ -1814,7 +1814,7 @@ class TestKstat:
         m1 = stats.moment(data, order=1)
         m2 = stats.moment(data, order=2)
         m3 = stats.moment(data, order=3)
-        xp_assert_close(xp.asarray((m1, m2, m3)), expected[:-1], atol=0.02, rtol=1e-2)
+        xp_assert_close(xp.stack((m1, m2, m3)), expected[:-1], atol=0.02, rtol=1e-2)
 
     @pytest.mark.filterwarnings("ignore:invalid value encountered in scalar divide")
     def test_empty_input(self, xp):
@@ -3247,11 +3247,11 @@ class TestCommonAxis:
         x = xp.asarray(rng.random((6, 7)))
 
         res = fun(x, **kwargs, axis=0)
-        ref = xp.asarray([fun(x[:, i], **kwargs) for i in range(x.shape[1])])
+        ref = xp.stack([fun(x[:, i], **kwargs) for i in range(x.shape[1])])
         xp_assert_close(res, ref)
 
         res = fun(x, **kwargs, axis=1)
-        ref = xp.asarray([fun(x[i, :], **kwargs) for i in range(x.shape[0])])
+        ref = xp.stack([fun(x[i, :], **kwargs) for i in range(x.shape[0])])
         xp_assert_close(res, ref)
 
         res = fun(x, **kwargs, axis=None)

--- a/scipy/stats/tests/test_stats.py
+++ b/scipy/stats/tests/test_stats.py
@@ -5175,8 +5175,8 @@ def test_ttest_ind(xp):
     # regression test
     tr = xp.asarray(1.0912746897927283)
     pr = xp.asarray(0.27647818616351882)
-    tr_2D = xp.asarray([tr, -tr])
-    pr_2D = xp.asarray([pr, pr])
+    tr_2D = xp.stack([tr, -tr])
+    pr_2D = xp.stack([pr, pr])
 
     rvs1 = xp.linspace(5, 105, 100)
     rvs2 = xp.linspace(1, 100, 100)
@@ -6017,8 +6017,8 @@ class TestTTestInd:
         tr_uneq_n = xp.asarray(0.66745638708050492)
         pr = xp.asarray(0.27647831993021388)
         pr_uneq_n = xp.asarray(0.50873585065616544)
-        tr_2D = xp.asarray([tr, -tr])
-        pr_2D = xp.asarray([pr, pr])
+        tr_2D = xp.stack([tr, -tr])
+        pr_2D = xp.stack([pr, pr])
 
         rvs3 = xp.linspace(1, 100, 25)
         rvs2 = xp.linspace(1, 100, 100)
@@ -6689,8 +6689,8 @@ class TestJarqueBera:
         res = stats.jarque_bera(x, axis=1)
         s0, p0 = stats.jarque_bera(x[0, :])
         s1, p1 = stats.jarque_bera(x[1, :])
-        xp_assert_close(res.statistic, xp.asarray([s0, s1]))
-        xp_assert_close(res.pvalue, xp.asarray([p0, p1]))
+        xp_assert_close(res.statistic, xp.stack([s0, s1]))
+        xp_assert_close(res.pvalue, xp.stack([p0, p1]))
 
         resT = stats.jarque_bera(x.T, axis=0)
         xp_assert_close(res.statistic, resT.statistic)
@@ -8237,7 +8237,7 @@ class TestCombinePvalues:
             pvaluess = xp.sort(xp.asarray(rng.uniform(0, 1, size=(m, n))), axis=0)
 
         combined_pvalues = xp.asarray([
-            stats.combine_pvalues(pvaluess[i, :], method=method)[1]
+            float(stats.combine_pvalues(pvaluess[i, :], method=method)[1])
             for i in range(pvaluess.shape[0])
         ])
         assert xp.all(combined_pvalues[1:] - combined_pvalues[:-1] >= 0)

